component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -683,4 +700,64 @@ private boolean validateJTIInCache(String jti, SignedJWT signedJWT, JWTCacheEntr
         }
         return true;
     }
+
+    /**
+     * Validate whether the request signing algorithm is configured for the application.
+     *
+     * @param signedJWT     The signed JWT.
+     * @param clientId      Client ID of the application.
+     * @return whether the request signing algorithm is configured for the application.
+     * @throws OAuthClientAuthnException OAuth Client Authentication Exception.
+     */
+    private boolean isValidSignatureAlgorithm(SignedJWT signedJWT, String clientId) throws OAuthClientAuthnException {
+
+        //   Obtain the signing algorithm used to sign the JWT in the request.
+        String requestSigningAlgorithm = signedJWT.getHeader().getAlgorithm().getName();
+        //   Obtain the signing algorithm configured for the application.
+        List<String> configuredSigningAlgorithms = getConfiguredSigningAlgorithm(clientId);
+        //   Mandating PS256 and ES256 as the JWT signing algorithms.
+        if (!(Constants.ALG_ES256.equals(requestSigningAlgorithm) ||
+                Constants.ALG_PS256.equals(requestSigningAlgorithm))) {
+            if (log.isDebugEnabled()) {
+                log.debug("FAPI unsupported signing algorithm " + requestSigningAlgorithm +
+                        " is used to sign the JWT.");
+            }
+            return false;
+        }
```

Comment:
Instead of checking against hardcoded sig algs lets check against the server configured fapi allowed sig algs as the configuration is in place to set the allowed algorithms.

Reply:
allowed sig alg => check get SP configured signing alg -> if not configured fallback to server configured values 
check against the allowed sig alg.

i feel above way of checking would be better, given we have checks in place to make sure sp configured sig alg is validated against the server allowed values when storing it in the sp

Reply:
Fixed in 6d4dc0d127ba6015f1896727ae8cd213e359417a